src/Nox.Ui.Blazor.Lib/Components/NoxTypes/EditTime.razor.cs

```diff
@@ -0,0 +1,221 @@
+﻿using Microsoft.AspNetCore.Components;
+using Nox.Types;
+using Nox.Ui.Blazor.Lib.Models;
+using System.Globalization;
+
+namespace Nox.Ui.Blazor.Lib.Components.NoxTypes;
+
+public partial class EditTime : ComponentBase
+{
+
+    #region Declarations
+
+    [Parameter]
+    public int? Hour { get; set; }
+
+    [Parameter]
+    public int? Minute { get; set; }
+
+    [Parameter]
+    public int? Second { get; set; }
+
+    [Parameter]
+    public int? Millisecond { get; set; }
+
+    [Parameter]
+    public bool DisplayMinute { get; set; } = true;
+
+    [Parameter]
+    public bool DisplaySecond { get; set; } = true;
+
+    [Parameter]
+    public bool DisplayMillisecond { get; set; } = false;
+
+    [Parameter]
+    public CultureInfo CultureInfo { get; set; } = CultureInfo.CurrentCulture;
+
+    [Parameter]
+    public string? TitleHour { get; set; }
+
+    [Parameter]
+    public string? TitleMinute { get; set; }
+
+    [Parameter]
+    public string? TitleSecond { get; set; }
+
+    [Parameter]
+    public string? TitleMillisecond { get; set; }
+
+    [Parameter]
+    public EventCallback<int?> HourChanged { get; set; }
+
+    public string? CurrentHourStr { get; set; }
+
+    [Parameter]
+    public EventCallback<int?> MinuteChanged { get; set; }
+
+    public string? CurrentMinuteStr { get; set; }
+
+    [Parameter]
+    public EventCallback<int?> SecondChanged { get; set; }
+
+    public string? CurrentSecondStr { get; set; }
+
+    [Parameter]
+    public EventCallback<int?> MillisecondChanged { get; set; }
+
+    public string? CurrentMillisecondStr { get; set; }
+
+    public Dictionary<int, string?>? HourSelectionList { get; set; } = new Dictionary<int, string?>();
+
+    public Dictionary<int, string?>? MinuteSelectionList { get; set; } = new Dictionary<int, string?>();
+
+    public Dictionary<int, string?>? SecondSelectionList { get; set; } = new Dictionary<int, string?>();
+
+    public Dictionary<int, string?>? MillisecondSelectionList { get; set; } = new Dictionary<int, string?>();
+
+    #endregion
+
+    /// <summary>
+    /// Handles initial loading
+    /// </summary>
+    protected override void OnInitialized()
+    {
```

Comment:
the same here, all this dictionaries should/can be static?

Reply:
resolved on commit